app/graphql/resolvers/generic.rb

```diff
@@ -0,0 +1,32 @@
+module Resolvers
+  class Generic
+    def initialize(type)
+      @type = type
+    end
+
+    def self.for(type)
+      new(type)
+    end
+
+    def record
+      return unless model_class
+      base_class.include(Resolvers::Concerns::Record)
+    end
+
+    def collection
+      return unless model_class
+      base_class.include(Resolvers::Concerns::Collection)
+    end
+
+    private
+
+    attr_reader :type
```

Comment:
The position is fine if it has to be kept `private`, but has it to be private?

Reply:
@dLobatog yeah, I really don't understand why we're defining a private `attr_reader`. I don't think that makes any sense, since anything within this class will already have access to the instance variable `@type`. You can `delegate :model_class, to: :@type`